burnman/seismic.py

```diff
@@ -222,45 +280,59 @@ def v_s(self, depth):
 
         return self._lookup(depth, self.table_vs)
 
+    def QK(self, depth):
+
+        return self._lookup(depth, self.table_Qk)
+
+    def QG(self, depth):
+
+        return self._lookup(depth, self.table_Qmu)
+
 
     def density(self, depth):
 
         return self._lookup(depth, self.table_density)
 
     def depth(self, pressure):
-        if pressure > max(self.table_pressure) or pressure < min(self.table_pressure)  :
-           raise ValueError, "Pressure outside range of PREM"
-        radius = np.interp(pressure, self.table_pressure[::-1], self.table_radius[::-1] )
-        return self.earth_radius - radius
+        if self.table_depth[0]:
```

Comment:
ok, the test_partitioning.py led to the depth routine in PREM being very
wrong (example_geotherms was broken too, but test.sh didn't catch that).
Otherwise test_partitioning looks wrong too, because it puts GPa into the
anderson geotherm instead of Pa. Not sure if the has anything to do with
you not reproducing it Bob.


On Wed, Jan 14, 2015 at 5:11 PM, Sanne Cottaar <sanne.cottaar@gmail.com>
wrote:

> Oops...
>
> On Wed, Jan 14, 2015 at 4:54 PM, Ian Rose <notifications@github.com>
> wrote:
>
>> In burnman/seismic.py
>> <https://github.com/geodynamics/burnman/pull/107#discussion-diff-22950520>
>> :
>>
>> >
>> >      def density(self, depth):
>> >
>> >          return self._lookup(depth, self.table_density)
>> >
>> >      def depth(self, pressure):
>> > -        if pressure > max(self.table_pressure) or pressure < min(self.table_pressure)  :
>> > -           raise ValueError, "Pressure outside range of PREM"
>> > -        radius = np.interp(pressure, self.table_pressure[::-1], self.table_radius[::-1] )
>> > -        return self.earth_radius - radius
>> > +        if self.table_depth[0]:
>>
>> Oh, sorry, I wasn't clear. I agree this is a useful error message. It
>> just looks like you are removing it.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/geodynamics/burnman/pull/107/files#r22950520>.
>>
>
>